src/commands/org/login/sfdx-url.ts

```diff
@@ -67,15 +75,21 @@ export default class LoginSfdxUrl extends AuthBaseCommand<AuthFields> {
 
   public async run(): Promise<AuthFields> {
     const { flags } = await this.parse(LoginSfdxUrl);
-    if (await this.shouldExitCommand(flags['no-prompt'])) return {};
 
-    const authFile = flags['sfdx-url-file'];
+    if (await this.shouldExitCommand(flags['no-prompt'])) return {};
 
-    const sfdxAuthUrl = authFile.endsWith('.json') ? await getUrlFromJson(authFile) : await readFile(authFile, 'utf8');
+    const sfdxUrlFile = flags['sfdx-url-file'];
+    const sfdxAuthUrl = flags['sfdx-url-stdin']
+      ? await read()
+      : sfdxUrlFile
+      ? sfdxUrlFile.endsWith('.json')
+        ? await getUrlFromJson(sfdxUrlFile)
+        : await readFile(sfdxUrlFile, 'utf8')
+      : null;
 
     if (!sfdxAuthUrl) {
       throw new Error(
```

Comment:
I know this was already an `Error`, but could we move the message to the messages file, and then convert this to be `throw messages.createError(...`